src/models/Factor.js

```diff
@@ -204,28 +205,30 @@ function (Okta, CookieUtil, factorUtil, BaseLoginModel) {
 
         var promise;
         // MFA_REQUIRED
-        if (transaction.getFactorById) {
-          promise = transaction
-          .getFactorById(this.get('id'))
-          .verifyFactor(data);
+        if (transaction.status === 'MFA_REQUIRED') {
+          var factor = _.findWhere(transaction.factors, {
+            id: this.get('id')
+          });
+          promise = factor.verify(data);
         }
 
         // MFA_CHALLENGE
-        else if (this.get('canUseResend') && transaction.resendByName) {
-          var firstLink = transaction.response._links.resend[0];
-          promise = transaction.resendByName(firstLink.name);
+        else if (this.get('canUseResend') && transaction.resend) {
+          var firstLink = transaction.data._links.resend[0];
+          promise = transaction.resend(firstLink.name);
```

Comment:
I noticed in the RecoveryChallengeController that you removed getting the firstLink - is this something we can do here?